module/applications/advancement/advancement-config.mjs

```diff
@@ -12,10 +12,10 @@ export default class AdvancementConfig extends FormApplication {
     super(advancement, options);
 
     /**
-     * The advancement being created or edited.
-     * @type {Advancement}
+     * The ID of the advancement being created or edited.
+     * @type {string}
      */
-    this.advancement = advancement;
+    this._advancementId = advancement.id;
```

Comment:
Should probably be hard-private unless we expect external code to inspect and modify this?